workflows/snakemake/Snakefile

```diff
@@ -1,53 +1,115 @@
-configfile: "./config.yaml"
-workdir: config["output_directory"]
+configfile: workflow.source_path("config.yaml")
+envvars: "NETCDF"
 
-envvars:
-    "NETCDF",
+import f90nml
+from pathlib import Path
 
-rule DEFAULT:
-    # First rule is executed by default. It requires output of the last rule to be present.
-    input: "finished.clean"
+rule ALL_EXPERIMENTS:
+    input: expand("{experiment}/finished.clean", experiment=config["experiments"].keys())
 
 rule CREATE_WORKDIR:
-    input: workflow.source_path("namelist.wps"), workflow.source_path("GEOGRID.TBL"), workflow.source_path("namelist.input"), workflow.source_path("wrf.job")
-    output: "namelist.wps", "GEOGRID.TBL", "namelist.input"
-    shell: "cp {input} ."
+    input:
+        wrf_job = workflow.source_path("wrf.job")
+    output:
+        "{experiment}/GEOGRID.TBL",
+    params:
+        geogrid_table = lambda wildcards: workflow.source_path(config['experiments'][wildcards.experiment]['geogrid_table']),
+    shell:
+        """
+        mkdir -p {wildcards.experiment}
+        cd {wildcards.experiment}
+        cp {input.wrf_job} .
+        cp {params.geogrid_table} GEOGRID.TBL
+        """
+
+rule UPDATE_NAMELISTS:
+    input:
+        namelist_wps = workflow.source_path("namelist.wps"),
+        namelist_input = workflow.source_path("namelist.input"),
+    output:
+        "{experiment}/namelist.wps",
+        "{experiment}/namelist.input"
+    run:
+        geog_data_res = config['experiments'][wildcards.experiment]['geog_data_res']
+        geog_data_path = config['experiments'][wildcards.experiment]['geog_data_path']
+        num_land_cat = config['experiments'][wildcards.experiment]['num_land_cat']
+        run_w2w = config['experiments'][wildcards.experiment]["run_w2w"]
+
+        # Read source namelist.wps with f90nml, make some changes, and save in work dir
+        wps_path = Path(wildcards.experiment, "namelist.wps")
+        nml = f90nml.read(input.namelist_wps)
+        nml["geogrid"]["geog_data_path"] = geog_data_path
+        nml["geogrid"]["geog_data_res"] = geog_data_res
+        nml.write(wps_path)
+
+        # Read source namelist.input, make some changes, and save in work dir
+        input_path = Path(wildcards.experiment, "namelist.input")
+        nml_input = f90nml.read(input.namelist_input)
+        nml_input["physics"]["num_land_cat"] = num_land_cat
+        nml_input["physics"]["use_wudapt_lcz"] = 1 if run_w2w else 0
+        nml_input.write(input_path)
+
 
 rule GEOGRID:
-    input: "namelist.wps", "GEOGRID.TBL"
-    output: "finished.geogrid"
+    input:
+        "{experiment}/namelist.wps",
+        geogrid = "{experiment}/GEOGRID.TBL"
+    output:
+        "{experiment}/finished.geogrid"
     shell:
         """
+        cd {wildcards.experiment}
+
+        ln -sf {input.geogrid} {config[wps_home]}
```

Comment:
This linking shouldn't be necessary. We're invoking WRF from the working directory `wildcards.experiment`, and it will look for the geogrid table in that directory (since we've set in namelist `opt_geogrid_tbl_path = '.')